### PR TITLE
Introducing World:SetSpawn() API and Lua binding

### PR DESCRIPTION
**Pull request by ElNounch**

### CODE REVIEW

**Comment by madmaxoft** [comment]
Looks good to me. I'd appreciate one more person reviewing this, though.

File: src/World.cpp [hunk 1]
```diff
@@ -324,6 +324,33 @@ void cWorld::SetNextBlockTick(int a_BlockX, int a_BlockY, int a_BlockZ)
...
+{
+		cIniFile IniFile;
+
+		IniFile.ReadFile(m_IniFileName);
+		IniFile.SetValueF("SpawnPosition", "X", a_X);
+		IniFile.SetValueF("SpawnPosition", "Y", a_Y);
+		IniFile.SetValueF("SpawnPosition", "Z", a_Z);
+		if (IniFile.WriteFile(m_IniFileName))
+		{
+			m_SpawnX = a_X;
```
Comment (bearbin) [inline src/World.cpp]: Should these be initialised even if the file cannot be written?

**Comment by bearbin** [comment]
Aside from my comment, LGTM.

File: src/World.cpp (hunk 1 above)
Comment (madmaxoft) [inline src/World.cpp]: I don't have an opinion in either way. Once you get a file write error, having a correct spawn coord is the least of your problems; perhaps a log message would be good in this case, though.

Comment (NiLSPACE) [inline src/World.cpp]: I think that could only cause confusion after a restart. *what!? the spawn is back to where it was! It was at the new place a minute ago.* Though of course they'd have to miss the warning log in the console for that as well.

Comment (ElNounch) [inline src/World.cpp]: That was the idea, if it ultimately failed (couldn't store information), don't look at all as if it worked.

**Comment by madmaxoft** [comment]
Ay, don't merge master, rebase onto it. Next time.

**Pull request merged**